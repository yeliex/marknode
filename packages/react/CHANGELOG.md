# @oh-markdown/react

## 1.1.0

### Minor Changes

- suppoer custom provider and react render
