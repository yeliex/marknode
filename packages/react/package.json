{
  "name": "@oh-markdown/react",
  "version": "1.0.5",
  "description": "Oh! Markdown~ react render",
  "main": "es/index.js",
  "type": "module",
  "typings": "es/index.d.ts",
  "files": [
    "es",
    "src",
    "LICENSE",
    "README.md"
  ],
  "scripts": {
    "clean": "rm -rf es",
    "build": "tsc -p tsconfig.json",
    "test:type": "tsc --noEmit"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/yeliex/oh-markdown.git"
  },
  "author": "yeliex <yeliex@yeliex.com>",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/yeliex/oh-markdown/issues"
  },
  "homepage": "https://github.com/yeliex/oh-markdown#readme",
  "dependencies": {
    "@mdx-js/react": "^2.2.1"
  },
  "devDependencies": {
    "oh-markdown": "^1.0.5"
  },
  "peerDependencies": {
    "react": ">=17.0.0"
  }
}
