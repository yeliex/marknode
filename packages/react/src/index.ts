export * from '@mdx-js/react';
export { default as MDComponent } from './MDComponent.js';
