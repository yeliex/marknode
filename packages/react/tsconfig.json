{
  "extends": "../../tsconfig.base.json",
  "compileOnSave": true,
  "compilerOptions": {
    "jsx": "react-jsx",
    "baseUrl": "./",
    "sourceRoot": "src",
    "outDir": "es"
  },
  "include": [
    "src"
  ]
}
