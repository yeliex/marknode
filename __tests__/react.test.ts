import { MDComponent } from '@oh-markdown/react';
import { Processor } from 'oh-markdown';
import { createElement } from 'react';
import { renderToStaticMarkup } from 'react-dom/server';

const md = `# Hello World {#hello,center}

## list {font-semibold,color=#777777}

{>color=red}
- 1 {color=blue}
- 2

center-test {center}

important {color=red,font-semibold}
multi-line
        `;

describe('Render Node Result', () => {
    const processor = new Processor();

    test('should render', async () => {
        const result = await processor.toNode(
            `# Hello World

## sub

- 1
- 2

`,
        );

        const str = renderToStaticMarkup(
            createElement(MDComponent, {
                nodes: result.data,
            }),
        );

        expect(str).toMatchSnapshot();
    });

    test('should render custom properties', async () => {
        const result = await processor.toNode(md);

        const str = renderToStaticMarkup(
            createElement(MDComponent, {
                nodes: result.data,
            }),
        );

        expect(str).toMatchSnapshot();
    });
});

describe('Render Html Result', () => {
    const processor = new Processor();

    test('should render', async () => {
        const result = await processor.toHtml(
            `# Hello World

## sub

- 1
- 2

`,
        );

        const str = renderToStaticMarkup(
            createElement(MDComponent, {
                nodes: result.data,
            }),
        );

        expect(str).toMatchSnapshot();
    });

    test('should render custom properties', async () => {
        const result = await processor.toHtml(md);

        const str = renderToStaticMarkup(
            createElement(MDComponent, {
                nodes: result.data,
            }),
        );

        expect(str).toMatchSnapshot();
    });
});
