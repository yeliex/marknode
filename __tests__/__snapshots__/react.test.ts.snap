// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Html Result should render 1`] = `"<script></script><h1 id="hello-world">Hello World</h1><h2 id="sub">sub</h2><ul><li>1</li><li>2</li></ul><script></script>"`;

exports[`Render Html Result should render custom properties 1`] = `"<script></script><h1 style="text-align: center" id="hello">Hello World</h1><h2 style="font-weight: 600; color: #777777" id="list">list</h2><ul style="color: red"><li style="color: blue">1</li><li>2</li></ul><p style="text-align: center">center-test</p><p><span style="color: red; font-weight: 600">important</span><br>multi-line</p><script></script>"`;

exports[`Render Node Result should render 1`] = `"<h1 id="hello-world">Hello World</h1><h2 id="sub">sub</h2><ul><li>1</li><li>2</li></ul>"`;

exports[`Render Node Result should render custom properties 1`] = `"<h1 style="text-align:center" id="hello">Hello World</h1><h2 style="font-weight:600;color:#777777" id="list">list</h2><ul style="color:red"><li style="color:blue">1</li><li>2</li></ul><p style="text-align:center">center-test</p><p><span style="color:red;font-weight:600">important</span><br/>multi-line</p>"`;
